pydm/tests/test_data_plugins_import.py

```diff
@@ -24,10 +24,12 @@ def test_plugin_directory_loading(qapp):
         handle.flush()
 
     try:
-        # Load plugins
+        # Load plugins. One of them will raise an exception during initialization. This will
+        # prevent it from being added to the available plugins, but the others should still load.
         load_plugins_from_path([cur_dir], 'foo.py')
         assert 'tst1' in plugin_modules
         assert 'tst2' in plugin_modules
+        assert 'fail' not in plugin_modules
```

Comment:
You might also consider taking a look at [`caplog`](https://docs.pytest.org/en/7.1.x/how-to/logging.html#caplog-fixture) to verify that the exception message was seen. That way you can be sure it was actually found, instantiated, and skipped over.

Reply:
Thanks for the caplog suggestion! Added here: 4260257ca7b13524ffc182a5cd7158be3672583d